tools/loadtesting/README.md

```diff
@@ -0,0 +1,51 @@
+# Mixer load testing tool
+
+This tool is meant for loadtesting mixer APIs in live clusters. It is based on [locust](https://docs.locust.io/en/stable/).
+
+The specific APIs that are called are defined by a [locustfile](https://docs.locust.io/en/stable/writing-a-locustfile.html), located under `tools/loadtesting/locustfiles`. 
+
+## Usage
+
+If the mixer instance to be tested lives as a k8s service that is not exposed to the internet, we can run a containerized version in the same cluster.
+
+1.  Deploy the pod into the same namespace as Mixer.
+
+```sh
+kubectl apply -f tools/loadtesting/pod.yaml --namespace=<mixer namespace>
+```
+
+2.  Make sure the pod is running with `kubectl -n <mixer namespace> get pod mixer-loadtester`. Output below means successful.
+
+```sh
+NAME                                           READY   STATUS    RESTARTS      AGE
+mixer-loadtester                               1/1     Running   0             0m
+```
+
+3.  Forward loadtester port to localhost.
+
+```sh
+kubectl -n <mixer namespace> port-forward mixer-loadtester 8089:8089
+```
+
+4.  Go to http://localhost:8089/ in browser. You should see the locust UI.
+5.  Click "New test", set the parameters(described below) and click "Start swarming".
+
+| Parameter | Definition |
+| :---: | :---: | 
+| Number of users  | Peak number of users. Note the behavior of each user is specified by a locustfile. 100 is a good default. |
+| Spawn rate  | Number of users added per second. This value is recommended to be <5. |
+| Host  | http:// + result of `kubectl -n <mixer namespace> get svc`. | 
+
+## Making changes to this tool
+
+1.  Set up the local environment.
+
+`virtualenv env && source env/bin/active && pip install -r requirements.txt`
+
+2.  After desired changes (most likely edit/add of a locustfile), you may want to change the default locust file inside `Dockerfile`.
```

Comment:
is it possible to not building a new image each time, but control it in the UI? So ideally the locustfile would have all tests..

Reply:
Unfortunately that's not provided out of the box. We can always make local changes and launch this tool locally, but it can't be used to loadtest k8s services that are not exposed to the internet.